### PR TITLE
Add tests and CI

### PR DESCRIPTION
**Pull request by BrightSkyz**
This commit adds tests for the library (both the standalone function and the middleware) alongside integration with Travis CI.

### CODE REVIEW

File: src/__tests__/utils/SharedTestUtils.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+};
+export type ExampleRequestResponse = {
+  status: number;
+  body: string;
+};
+
+export function signRequestWithKeyPair(body: string, privateKey: Uint8Array): SignedRequest {
+  const timestamp = String(new Date().getTime());
+  const signature = Buffer.from(nacl.sign.detached(
+    Uint8Array.from(Buffer.from(Buffer.concat([Buffer.from(timestamp), Buffer.from(body)]).toString('hex'), 'hex')),
```
Comment (typpo) [inline src/__tests__/utils/SharedTestUtils.ts:41]: I think we can simplify this slightly by not converting to/from hex in the intermediate steps

File: src/__tests__/utils/SharedTestUtils.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  data: {
+    id: '787053080478613554',
+    name: 'test'
+  }
+});
+
+// Generate a "valid" keypair
+export const generatedValidKeyPair = nacl.sign.keyPair();
+// Generate an "invalid" keypair
+export const generatedInvalidKeyPair = nacl.sign.keyPair();
```
Comment (typpo) [inline src/__tests__/utils/SharedTestUtils.ts:26]: nit: I think we can drop the `generated` prefix

File: src/__tests__/verifyKey.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  generatedInvalidKeyPair,
+  generatedValidKeyPair,
+  pingRequestBody,
+  signRequestWithKeyPair,
+} from './utils/SharedTestUtils';
+
+describe('verify key method', () => {
+  it('valid', () => {
+    // Sign and verify a valid request
+    const signedRequest = signRequestWithKeyPair(pingRequestBody, generatedValidKeyPair.secretKey);
```
Comment (typpo) [inline src/__tests__/verifyKey.ts:12]: Not a blocker, but perhaps you want to parameterize these tests to also try `applicationCommandRequestBody`?

File: src/__tests__/verifyKeyMiddleware.ts [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+import { AddressInfo } from 'net';
+import {
+  applicationCommandRequestBody, generatedInvalidKeyPair,
+  generatedValidKeyPair,
+  pingRequestBody,
+  sendExampleRequest,
+  signRequestWithKeyPair,
+} from './utils/SharedTestUtils';
+import * as http from 'http';
+import DoneCallback = jest.DoneCallback;
```
Comment (typpo) [inline src/__tests__/verifyKeyMiddleware.ts:12]: nit: we don't have import order linter in this project, but let's move the 3rd party deps above project deps

File: src/__tests__/verifyKeyMiddleware.ts [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+  it('valid', async (done: DoneCallback) => {
+    // Sign and verify a valid request
+    const signedRequest = signRequestWithKeyPair(pingRequestBody, generatedValidKeyPair.secretKey);
+    const exampleRequestResponse = await sendExampleRequest(exampleInteractionsUrl, {
+      'x-signature-ed25519': signedRequest.signature,
+      'x-signature-timestamp': signedRequest.timestamp,
+      'content-type': 'application/json'
+    }, signedRequest.body)
+    expect(exampleRequestResponse.status).toBe(200);
+    done();
```
Comment (typpo) [inline src/__tests__/verifyKeyMiddleware.ts:53]: Isn't `done()` unnecessary in an async test?  https://jestjs.io/docs/en/asynchronous#asyncawait

**Review by typpo — CHANGES_REQUESTED** [review]
This is looking pretty good and I can't thank you enough for creating these tests!  It will make it a lot easier to update this library in the future.

Just a handful of minor comments :)

File: src/__tests__/verifyKeyMiddleware.ts (hunk 5 above)
Comment (BrightSkyz) [reply src/__tests__/verifyKeyMiddleware.ts:53]: The reason for this was that it is using callbacks. I don't exactly remember why I chose this method but if you would like, I could convert it away from using callbacks. https://jestjs.io/docs/en/asynchronous#callbacks

**Pull request merged**